### PR TITLE
[stable/prometheus-operator] Fix duplicate prometheus server scrape when per-replica service enabled

### PR DESCRIPTION
**Pull request by shaunmansell**
#### What this PR does / why we need it:

If the Prometheus per-replica service is enabled along with
`prometheus.serviceMonitor.selfMonitor`, then the match conditions on the
serviceMonitor will also match the per-replica service meaning each
Prometheus server replica will be scraped twice.

This updates the main Prometheus service resource to include a new `self-monitor`
label that will be set to the value of `prometheus.serviceMonitor.selfMonitor`. The
label matching on the Prometheus serviceMonitor has been updated to include a
new condition where the new `self-monitor` label must be present with a the value set
to `true`.

Another I thought we could solve this is to hard code a relabeling drop action on the prometheus `serviceMonitor` resource that drops any metrics from endpoints that included the `statefulset.kubernetes.io/pod-name` pod label.

e.g.
```
relabelings:
  - source_labels: [__meta_kubernetes_pod_label_statefulset_kubernetes_io_pod_name]
    regex: .+
    action: drop
```

I'm sure there's other ways and am happy to adapt this PR to suit whatever the collective suggest.

Also fixes a namespace reference issue for the per-replica ingress resource from the changes in #17689

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by shaunmansell** [comment]
Currently the ci values do not enable the per-replica service or ingress resources since they're off by default and that state was carried over to the test values file.

This will mean the automated testing never applies config related to those chart features and we won't see failures for future changes that may impact them.

Should we flip `prometheus.servicePerReplica` and `prometheus.ingressPerReplica` to `true` to have them created during testing and try catch failures earlier?

This would have helped avoid some of the issues introduced with #17689

**Comment by zanhsieh** [comment]
@shaunmansell 
I think we should temporary turn off `selfMonitor` as `False` in values.yaml, or might think a way to limit prometheus scrapping all namespaces. This is the reason why pull-charts-e2e test failed (inquiry too many unrelated charts in other PRs causing timeout). Check pull-charts-e2e -> Details in below to see the detail.

**Comment by shaunmansell** [comment]
I think they failure is being caused by node capacity issues on scheduling the `node-exporter` pods. Looking through the test logs I can see:

```
I1015 09:42:28.432] Events:
I1015 09:42:28.433]   Type     Reason             Age                   From                Message
I1015 09:42:28.433]   ----     ------             ----                  ----                -------
I1015 09:42:28.433]   Normal   NotTriggerScaleUp  5m3s (x6 over 9m26s)  cluster-autoscaler  pod didn't trigger scale-up (it wouldn't fit if a new node is added): 1 node(s) didn't match node selector, 1 node(s) didn't have free ports for the requested pod ports
I1015 09:42:28.434]   Warning  FailedScheduling   13s (x45 over 10m)    default-scheduler   0/3 nodes are available: 2 node(s) didn't match node selector, 3 node(s) didn't have free ports for the requested pod ports.
I1015 09:42:28.434]   Normal   NotTriggerScaleUp  4s (x56 over 10m)     cluster-autoscaler  pod didn't trigger scale-up (it wouldn't fit if a new node is added): 1 node(s) didn't have free ports for the requested pod ports, 1 node(s) didn't match node selector
```

This is reported for all 3 `node-exporter` pods that are from the inclusion of [stable/node-exporter] as a chart dependency.

The self monitor configuration has always been enabled and my changes here have not altered that. Other recent PRs like #17918 have passed testing with the same self monitor configuration. More recent PRs like #17975 are impacted by the same capacity issue.

To fix this in the immediate term I can update `nodeExporter.enabled` to `false` in the test values, though it seems that it would be better to sort out the cluster capacity issues so that node-exporter testing is always done.

Who manages the test cluster capacity?

Should I proceed with disabling the node-exporter in the test values?

**Comment by shaunmansell** [comment]
@zanhsieh disabling the node-exporter chart in the test values fixes the test failures, though this doesn't seem like the best solution for the long term.

**Comment by zanhsieh** [comment]
@vsliouniaev 
Would you mind to LGTM this chart?

@shaunmansell 
Yeap, I agreed. So far I cannot see the configuration to let node-exporter only exports some labeled nodes; that's the best thing we can do.

File: stable/prometheus-operator/ci/test-values.yaml [hunk 1]
```diff
@@ -885,7 +885,7 @@ kube-state-metrics:
 ## Deploy node exporter as a daemonset to all nodes
 ##
 nodeExporter:
-  enabled: true
+  enabled: false
```
Comment (vsliouniaev) [inline stable/prometheus-operator/ci/test-values.yaml]: This change should not be included in the PR, it prevents testing configurations of the node-exporter sub-chart

**Review by vsliouniaev — CHANGES_REQUESTED** [review]

File: stable/prometheus-operator/ci/test-values.yaml (hunk 1 above)
Comment (shaunmansell) [reply stable/prometheus-operator/ci/test-values.yaml]: I have re-enabled the node-exporter in the test values however this is likely to break the tests again due to the pod placement issues on the test cluster.

If the chart test is failing will that block merging this change?

**Comment by vsliouniaev** [comment]
Could you please update from latest, the tests should work
/test all

**Comment by zanhsieh** [comment]
@shaunmansell 
Would you mind to resolve conflict file please?
stable/prometheus-operator/Chart.yaml
Usually it's just version bump.

**Comment by shaunmansell** [comment]
@vsliouniaev I've pulled in master and fixed the chart version conflict and see all tests are passing so this should be good to go
